Array type-checking failing

What @jaclark5 noticed in #109 has also been happening upstream for a few days:

```
yammbs/_base/array.py:25: error: Incompatible return value type (got "list[str]", expected "list[float]")  [return-value]
```

https://github.com/openforcefield/yammbs/actions/workflows/ci.yaml?query=event%3Aschedule

I can't obviously tie this to any new upstream releases

--- a/setup.cfg
+++ b/setup.cfg
@@ -49,6 +49,7 @@ warn_unused_configs = True
 warn_unused_ignores = True
 warn_incomplete_stub = True
 show_error_codes = True
+exclude = yammbs/_tests/
 
 [mypy-openeye]
 ignore_missing_imports = True

--- a/yammbs/_base/array.py
+++ b/yammbs/_base/array.py
@@ -21,7 +21,7 @@ def _strip_units(val: list[float] | Quantity | NDArray) -> NDArray:
     return numpy.asarray(unitless_val).reshape((-3, 3))
 
 
-def _array_serializer(val: NDArray, nxt) -> list[float]:
+def _array_serializer(val: NDArray[numpy.float64], nxt) -> list[float]:
     return val.flatten().tolist()
 
 

